code/modules/admin/antag_panel.dm

```diff
@@ -186,7 +186,8 @@ GLOBAL_VAR(antag_prototypes)
 		var/uplink_info = "<i><b>Uplink</b></i>:"
 		var/datum/component/uplink/U = find_syndicate_uplink()
 		if(U)
-			uplink_info += "<a href='?src=[REF(src)];common=takeuplink'>take</a>"
+			if(!U.uplink_handler.has_objectives)
+				uplink_info += "<a href='?src=[REF(src)];common=takeuplink'>take</a>"
```

Comment:
if you remove someone's antag datum, are you still unable to take this away? Or does removing the antag datum remove the uplink handler's var?

Reply:
fixed